flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -37,51 +37,58 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.fail;
 
 /** {@link ChannelStateWriterImpl} lifecycle tests. */
-public class ChannelStateWriterImplTest {
+class ChannelStateWriterImplTest {
     private static final long CHECKPOINT_ID = 42L;
     private static final String TASK_NAME = "test";
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testAddEventBuffer() throws Exception {
+    @Test
+    void testAddEventBuffer() {
 
         NetworkBuffer dataBuf = getBuffer();
         NetworkBuffer eventBuf = getBuffer();
         eventBuf.setDataType(Buffer.DataType.EVENT_BUFFER);
-        try {
-            runWithSyncWorker(
-                    writer -> {
-                        callStart(writer);
-                        writer.addInputData(
-                                CHECKPOINT_ID,
-                                new InputChannelInfo(1, 1),
-                                1,
-                                ofElements(Buffer::recycleBuffer, eventBuf, dataBuf));
-                    });
-        } finally {
-            assertTrue(dataBuf.isRecycled());
-        }
+        assertThatThrownBy(
+                        () -> {
+                            try {
+                                runWithSyncWorker(
+                                        writer -> {
+                                            callStart(writer);
+                                            writer.addInputData(
+                                                    CHECKPOINT_ID,
+                                                    new InputChannelInfo(1, 1),
+                                                    1,
+                                                    ofElements(
+                                                            Buffer::recycleBuffer,
+                                                            eventBuf,
+                                                            dataBuf));
+                                        });
+                            } finally {
+                                assertThat(dataBuf.isRecycled()).isTrue();
+                            }
+                        })
+                .isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
-    public void testResultCompletion() throws IOException {
+    void testResultCompletion() throws IOException {
         ChannelStateWriteResult result;
         try (ChannelStateWriterImpl writer = openWriter()) {
             callStart(writer);
             result = writer.getAndRemoveWriteResult(CHECKPOINT_ID);
-            assertFalse(result.resultSubpartitionStateHandles.isDone());
-            assertFalse(result.inputChannelStateHandles.isDone());
+            assertThat(result.resultSubpartitionStateHandles.isDone()).isFalse();
+            assertThat(result.inputChannelStateHandles.isDone()).isFalse();
```

Comment:
```suggestion
            assertThat(result.resultSubpartitionStateHandles).isNotDone();
            assertThat(result.inputChannelStateHandles).isNotDone();
```
assertJ supports some `CompletableFuture` methods